homeassistant/components/tradfri/light.py

```diff
@@ -195,6 +194,18 @@ def hs_color(self):
             if hue is not None and sat is not None:
                 return hue, sat
 
+    def _get_supported_features(self, device):
```

Comment:
Can we just inline this functionality into init method? The device is available there, right?

Reply:
It's enough if we set the features in init method and not update after that I'm thinking.

Reply:
I tried before proposing this change, but got this error (`init` method below):

```
Traceback (most recent call last):
  File "/home/patrik/Repos/hass/home-assistant/homeassistant/helpers/entity_platform.py", line 245, in async_add_entities
    for entity in new_entities
  File "/home/patrik/Repos/hass/home-assistant/homeassistant/helpers/entity_platform.py", line 242, in <listcomp>
    self._async_add_entity(
  File "/home/patrik/Repos/hass/home-assistant/homeassistant/components/tradfri/light.py", line 45, in <genexpr>
    async_add_entities(TradfriLight(light, api, gateway_id) for light in lights)
  File "/home/patrik/Repos/hass/home-assistant/homeassistant/components/tradfri/light.py", line 155, in __init__
    self.supported_features = self._get_supported_features(self._device)
AttributeError: can't set attribute
```

```
    def __init__(self, device, api, gateway_id):
        """Initialize a Light."""
        super().__init__(device, api, gateway_id)
        self._unique_id = f"light-{gateway_id}-{device.id}"
        self._hs_color = None

        self.supported_features = self._get_supported_features(self._device)

        self._refresh(device)

```

Reply:
`supported_features` is a property in the `Entity` class that we inherit from. Can such properties be overwritten in `__init__` or must an overwriting `@property` be defined?

Reply:
Define a private attribute, `self._features`, that we return in the property.

Reply:
And remove the method to get the features. Just do that in init method.

Reply:
```py
features = SUPPORTED_LIGHT_FEATURES
if device.light_control.can_set_dimmer:
    features |= SUPPORT_BRIGHTNESS
if device.light_control.can_set_color:
    features |= SUPPORT_COLOR
if device.light_control.can_set_temp:
    features |= SUPPORT_COLOR_TEMP
		
self._features = features
```

Reply:
Oh, gotcha. Pushing change.